### PR TITLE
Drop default value for nodepool .management.replace

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Otherwise default meaningless applies even when UpgradeType=inplace

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: api/v1alpha1/nodepool_types.go [hunk 1]
```diff
@@ -263,7 +263,6 @@ type NodePoolManagement struct {
 	// Replace is the configuration for rolling upgrades.
 	//
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:default={strategy: "RollingUpdate", rollingUpdate: {maxSurge: 1, maxUnavailable: 0 }}
```
Comment (alvaroaleman) [inline api/v1alpha1/nodepool_types.go:266]: what will the capi machinedeployment controller do if these are unset?

Comment (alvaroaleman) [reply api/v1alpha1/nodepool_types.go:266]: Okay, then the `optional` is wrong now?

Comment (enxebre) [reply api/v1alpha1/nodepool_types.go:266]: no because it can be empty when UpgradeType=InPlace

Comment (enxebre) [reply api/v1alpha1/nodepool_types.go:266]: @alvaroaleman Let's proceed with this PR and address the webhook separately following up to https://github.com/openshift/hypershift/pull/1346 so the validation logic can be reused between backend and webhook.

Comment (davidvossel) [reply api/v1alpha1/nodepool_types.go:266]: the maxSurge and maxUnavailable fields themselves still document default values

https://github.com/openshift/hypershift/blob/eaea63a0759260318fbb0ea35d24f9243223aa7f/api/v1alpha1/nodepool_types.go#L218

https://github.com/openshift/hypershift/blob/eaea63a0759260318fbb0ea35d24f9243223aa7f/api/v1alpha1/nodepool_types.go#L239

were these default values expected to come from this kubebuilder tag that's being removed?

Comment (enxebre) [reply api/v1alpha1/nodepool_types.go:266]: Thanks David, added the defaults for those.

Comment (davidvossel) [reply api/v1alpha1/nodepool_types.go:266]: So now what happens if someone sets the upgrade type to `replace` but doesn't explicilty set the `replace` field? Will it get defaulted somehow. What would be the expectation?

Comment (enxebre) [reply api/v1alpha1/nodepool_types.go:266]: It must fail since those are required fields, we'll add that defaulting to the optional webhook though.

**Pull request closed without merging**